azkaban-web-server/src/main/java/azkaban/flowtrigger/plugin/FlowTriggerDependencyPluginManager.java

```diff
@@ -47,19 +50,30 @@
   public static final String PRIVATE_CONFIG_FILE = "private.properties";
   public static final String DEPENDENCY_CLASS = "dependency.class";
   public static final String CLASS_PATH = "dependency.classpath";
-
   private static final Logger logger = LoggerFactory
       .getLogger(FlowTriggerDependencyPluginManager.class);
-
   private final String pluginDir;
-
   private final Map<String, DependencyCheck> dependencyTypeMap;
+  private final ClassLoader prevClassLoader;
 
   @Inject
-  public FlowTriggerDependencyPluginManager(final String pluginDir) throws
-      FlowTriggerDependencyPluginException {
+  public FlowTriggerDependencyPluginManager(final String pluginDir)
+      throws FlowTriggerDependencyPluginException {
     this.dependencyTypeMap = new ConcurrentHashMap<>();
     this.pluginDir = pluginDir;
+    this.prevClassLoader = Thread.currentThread().getContextClassLoader();
+  }
+
+  private File[] getFilesMatchingPath(final String path) {
```

Comment:
How will this method handle cases like:
`lib/gobblin-*`
?

Reply:
that's good question, actually it will only support dir/*.
Since user shouldn't upload the jars they don't want to import
the reason for supporting dir/* only is to provide another packaging option which let user upload a dir of all required jars instead of one fat jar.

Reply:
Can you leave a comment mentioning this? I could see somebody (either a dev or a user) getting confused in the future because there's certain expectations around the usage of `*` characters.

Reply:
thanks, added the comment